### PR TITLE
Support Android in Connection Spec Tests

### PR DESCRIPTION
**Pull request by yschimke**
Ciphers available changes on Android

### CODE REVIEW

**Comment by yschimke** [comment]
I can't land this in okhttp main tests because of Build dependency from Android.  So generally getting advice on the diff.

File: okhttp/src/main/kotlin/okhttp3/internal/platform/Platform.kt [hunk 1]
```diff
@@ -273,5 +274,13 @@ open class Platform {
       }
       return result.readByteArray()
     }
+
+    fun androidSdkVersion(): Int? {
```
Comment (swankjesse) [inline okhttp/src/main/kotlin/okhttp3/internal/platform/Platform.kt:278]: move to PlatformRuleTest?

File: okhttp/src/test/java/okhttp3/ConnectionSpecTest.java [hunk 2]
```diff
@@ -224,14 +227,31 @@
...
-        CipherSuite.TLS_RSA_WITH_RC4_128_MD5.javaName(),
+        CipherSuite.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256.javaName(),
+        CipherSuite.TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA.javaName(),
     });
 
     applyConnectionSpec(tlsSpec, sslSocket, false);
-    assertThat(sslSocket.getEnabledCipherSuites()).containsExactly(
-        CipherSuite.TLS_RSA_WITH_RC4_128_SHA.javaName(),
-        CipherSuite.TLS_RSA_WITH_RC4_128_MD5.javaName());
+    if (Platform.Companion.isAndroid()) {
```
Comment (swankjesse) [inline okhttp/src/test/java/okhttp3/ConnectionSpecTest.java:235]: Rad!

**Review by swankjesse — APPROVED** [review]
Nice

**Pull request merged**